assets/js/components/adminbar/AdminBarWidgets.test.js

```diff
@@ -53,14 +53,14 @@ describe( 'AdminBarWidgets', () => {
 	} );
 
 	it( 'should render the Admin Bar Widgets, including the Activate Analytics CTA', async () => {
-		const { container, getByText, waitForRegistry } = render(
+		const { container, getByText, waitForResolvingPromise } = render(
 			<AdminBarWidgets />,
 			{
 				registry,
 			}
 		);
 
-		await waitForRegistry();
+		await waitForResolvingPromise();
```

Comment:
Some resolvers were still asynchronously resolving after `waitForRegistry`, causing an error to be thrown due to the component updating outside of an `act`. Waiting for active Promises to resolve, via awaiting a new Promise which resolves in a timeout, solves this.

Reply:
In this situation I think it's still appropriate to wait for the registry as before. Ideally we could update `waitForRegistry` to wait for any remaining async updates but I realize that may not be possible.

Reply:
I've taken a different approach to the waits, having added a wrapper to `setTimeout` that allows timeouts to be tracked and therefore waited for in a more deterministic manner.

Then created a new helper, `waitForRegistryAndTimeouts` which wraps both `waitForRegistry` and the new `waitForTimeouts` utility, and used this where `waitForResolvingPromise` was previously being used.